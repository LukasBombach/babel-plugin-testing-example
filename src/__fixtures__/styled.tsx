import React from "react";
import styled from "passerine";

const StyledComponent = styled.div({
  color: "red",
  background: "blue",
});

export const Component = () => <StyledComponent />;
