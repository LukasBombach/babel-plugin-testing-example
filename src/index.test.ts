import { transformAsync } from "@babel/core";
import plugin from ".";

import type { BabelFileResult } from "@babel/core";

async function transform(code: string): Promise<BabelFileResult | null> {
  return await transformAsync(code, {
    filename: "test.tsx",
    presets: [() => ({ plugins: [plugin] })],
  });
}

test("simple example", async () => {
  const source = `
import React from "react";
import styled from "passerine";

const StyledComponent = styled.div({
  color: "red",
  background: "blue",
});

export const Component = () => <StyledComponent />;
`;

  const { code } = await transform(source);
  expect(code).toMatchSnapshot();
});
