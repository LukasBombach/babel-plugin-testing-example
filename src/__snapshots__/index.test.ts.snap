// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`simple example 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;

var _react = _interopRequireDefault(require(\\"react\\"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const StyledComponent = props => _react.default.createElement(\\"div\\", {
  \\"className\\": \\"p12p\\"
}, {
  \\"props\\": \\"children\\"
});

const Component = () => <StyledComponent />;

exports.Component = Component;"
`;
